devtools/gradle/src/main/java/io/quarkus/gradle/builder/QuarkusModelBuilder.java

```diff
@@ -402,6 +413,25 @@ private void addDevModePaths(final DependencyImpl dep, ResolvedArtifact a, Proje
         }
     }
 
+    private void addSubstitutedProject(final DependencyImpl dep, File projectFile, LaunchMode mode) {
```

Comment:
Just curious, have you tried to measure how long this method takes to execute? Is it seconds?

Reply:
I just ran some really basics benchmarks, it take 2 to 4 seconds..

Reply:
Sorry, do you mind not using Optional in this case? Making includeBuild return boolean will make it look simpler and perform better.

Reply:
Oh yes for sure. I'm trying to understand why the test fails in ci and passes on my machine. I will remove the optional once everything is ok.

Reply:
> I just ran some really basics benchmarks, it takes 2 to 4 seconds..

Just to clarify, was it a restart/reload that was taking 2 to 4 seconds in dev mode?

Reply:
It's for the first quarkusDev start. Reload is not impacted, and thanks to daemons/caches, next restart are not really impacted.

Reply:
Right, that's good. Thanks!